broadinstitute/single_cell_portal_core
# Update `react-scripts` to fix alert about `immer` (SCP-3893)

This fixes a [false-positive critical security vulnerability in `immer`](https://github.com/broadinstitute/single_cell_portal_core/security/dependabot/yarn.lock/immer/open).  That was a dependency of an old version of `react-scripts`, which we use in development -- and not production.  Nevertheless, the dependency can't easily be moved from `dependencies` to `devDependencies`.  Details: facebook/create-react-app#10411.

I drafted a more comprehensive set of dependency updates, but those caused deeply cryptic problems.  This is a minimal change that fixes the most-reportedly-severe error.

To test:
* Pull branch
* `yarn install --force`
* Restart `rails s`, verify no errors upon startup
* In another terminal, restart `bin/webpack-dev-server`, verify no errors upon startup
* In local SCP, navigate to a study that has visualizations
* Save a change to a JavaScript file, e.g. open `ScatterPlot.js`, add `console.log('ok')` atop `RawScatterPlot`
* In `webpack-dev-server` terminal, verify your change triggered a recompilation
* In local SCP, open DevTools console, verify "ok" was printed to console
* Revert change to `ScatterPlot.js`

This relates to SCP-3893.
